File: 백준/Gold/26266. 비즈네르 암호 해독/비즈네르 암호 해독.swift
import Foundation

@inline(__always)
func getAscii(_ c: Character) -> Int {
    return Int(c.asciiValue!) - 64
}

func getPrimaryNumbers(_ length: Int) -> [Int] {
    var result = Set<Int>()
    for i in 1...Int(Double(length).squareRoot()) {
        if length % i == 0 {
            result.insert(i)
            result.insert(length / i)
        }
    }
    return result.sorted()
}

func solution() {
    guard let plainLine = readLine(),
          let cipherLine = readLine() else { return }

    let plain = Array(plainLine)
    let cipher = Array(cipherLine)
    let length = plain.count

    let key: [Int] = zip(plain, cipher).map {
        let d = getAscii($1) - getAscii($0)
        return d > 0 ? d : d + 26
    }

    for p in getPrimaryNumbers(length) {
        let pattern = Array(key[0..<p])
        var isMatch = true

        for (i, k) in key.enumerated() {
            if pattern[i % p] != k {
                isMatch = false
                break
            }
        }

        if isMatch {
            let result = pattern.compactMap { UnicodeScalar($0 + 64) }.map { String($0) }.joined()
            print(result)
            break
        }
    }
}

solution()

File: 백준/Gold/26266. 비즈네르 암호 해독/비즈네르 암호 해독.py
from sys import stdin

readline = stdin.readline

def getNumber(ch):
    return ord(ch) - ord("A") + 1

def getString(lst):
    return "".join(chr(x + ord("A") - 1) for x in lst)

def getPrimaryNumberList(n):
    result = set()
    for i in range(1, int(n ** 0.5) + 1):
        if n % i == 0:
            result.add(i)
            result.add(n // i)
    return sorted(result)

plain = readline().rstrip()
cipher = readline().rstrip()
length = len(plain)

diffs = [
    (getNumber(c) - getNumber(p)) % 26 or 26
    for p, c in zip(plain, cipher)
]

for l in getPrimaryNumberList(length):
    segment = getString(diffs[:l])
    valid = True
    for i in range(0, length, l):
        if getString(diffs[i:i+l]) != segment:
            valid = False
            break
    if valid:
        print(segment)
        break